### DISCUSSION TITLE
Intégration de l'API orange-money-burkina-sdk de fasodev

### DISCUSSION THREAD

**Discussion by AbdoulNhak** [faso-dev/orange-money-burkina-sdk#27] [Q&A]
Salut chers Auteurs, je voudrais utiliser votre API pour intégrer le moyen de paiement orange sur une plateforme et je ne sais pas trop comment m'y prendre.
J'ai entamé le processus de partenariat avec orange mais depuis là ça traine

**Comment by faso-dev** [comment]
Bonsoir Mr @AbdoulNhak
Pour l'intégration de l'api Orange Money, rien de bien compliqué. Prenez attache avec la structure Orange Money Burkina afin d'acquérir vos identifiants d'accès à l'api. Une fois fait vous pouvez suivre la documentation du package.

Si vous avez déjà vos identifiants, vous pouvez nous en dire plus sur votre problème.

Merci

**Comment by AbdoulNhak** [comment]
Oui bonsoir à vous, j'ai déjà mes identifiants, j'ai aussi lu la
documentation de votre API mais je ne me retrouve toujours pas.
C'est concernant la partie sur l'envoi de la requête XML

**Reply by faso-dev** [reply]
Vous n'en avez rien plus besoin de faire le XML vous même. C'est déjà gérer.

Votre projet doit tourner sur PHP pour pouvoir utiliser le package.

Également disposer de composer installer.

Puis vous le reste, c'est de suivre l'exemple donné sur la doc.

**Comment by AbdoulNhak** [comment]
Bonjour, j'utilise Laravel et j'ai installé votre API, si vous avez
peut-être une vidéo de description de la suite du processus j'essaierai de
suivre ça pour voir comment avancer

**Comment by AbdoulNhak** [comment]
Après avoir installé votre API, qu'est ce que je dois faire comme
manipulation ?

**Reply by faso-dev** [reply]
```php
try {
    $orangeApi = (new OrangeMoneyAPI("username", "password", "merchant_number"))
                ->withTransactionData(TransactionData::from('client_number', 'payment_amount', 'otp_code'))
                ->withCustomReference("123456778") //optionnal
                ->useProdApi() // for production
                ->useDevApi() // for developement
                ->withoutSSLVerification() //if you have any troubleshoot with ssl verifcation(not recommended)
    ;
    $response = (new PaymentSDK($orangeApi))->handlePayment();
   dd( $response->getTransactionId());
} catch (TransactionException $exception) {
          dd$exception->getCode())";
}
```

Le username est fourni par Orange, le password également.
Dans le mail contenant tes identifiants fourni par Orange, ils vont te spécifier le merchant number aussi.

Ennsuite le client_number est le sécond numero de test fourni pas Orange que tu as réçu également. Ce numéro contient une somme permettant de faire les tests. 
Le payment_amount est le montant de votre Transaction de teste (200, 500, 1500, 150, etc.) .

Le code OTP est le code réçu une fois que tu as généré avec le code ussd réçu par email.

J'espère que vous vous en sortirez avec ces informations.

Si vous avez besoin d'une assistance privée, merci de me joindre par mail.

**Reply by saifonglee** [reply]
Je me permet de me joindre a cette discussion. J'ai egalement installe votre Package et j'ai voulu l'utiliser dans mon projet. Mais j'ai toujour l'erreur suivante: 
`cURL error 28: Failed to connect to testom.orange.bf port 9008 after 129221 ms: Connection timed out (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://testom.orange.bf:9008/payment`.

J'ai fai quelques recherches et il s'avere qu'il s'agit d'une erreur qui pourrais etre en rapport avec les variables `connect_timeout`, `read_timeout` et `timeout` dont les valeurs devraient etre vu a la hausse. chose que j'ai effectivement modifer mais toujours pareille.

puis-je beneficier d'une assisanstance? 
Merci d'avance.

**Reply by faso-dev** [reply]
Bonsoir, si vous avez cette erreur, cela vient de Orange qui mets trop de temps à répondre.

Le mieux c'est de les contacter pour avoir plus de réponses.

Même si tu augmente les connect timeout et autres cela ne va changer grande chose y'a que le serveur de Orange Burkina mettra du temps à répondre.

**Comment by AbdoulNhak** [comment]
Bonsoir merci beaucoup pour cette assistance, j'ai pu effectué les test.
S'il ya un besoin je vous joindrai sur votre mail privé si je peux en
profiter vous le solliciter.
Merci et bonne suite de soirée..

**Comment by AbdoulNhak** [comment]
Salut !! je reviens encore pour vous déranger
Au fait j'ai déployé ma solution et je rencontre cette erreur
:""production.ERROR:
Call to undefined method Fasodev\Sdk\OrangeMoneyAPI::withTransactionData()
{"userId":8,"exception":"[object] (Error(code: 0): Call to undefined method
Fasodev\\Sdk\\OrangeMoneyAPI::withTransactionData() ""

Pouvez-vous m'aider svp

**Reply by faso-dev** [reply]
Cest ::withTransactionData mais ->withTransactionData

```

 (new OrangeMoneyAPI("username", "password", "merchant_number"))
                ->withTransactionData(TransactionData::from('client_number', 'payment_amount', 'otp_code'))
                ->withCustomReference("123456778") //optionnal
                ->useProdApi() // for production
                ->useDevApi() // for developement
                ->withoutSSLVerification() //if you have any troubleshoot with ssl verifcation(not recommended)
    ;

````

**Reply by AbdoulNhak** [reply]
Pourtant en local tout marche très bien, vraiment je ne comprends rien, Pensez-vous que je devrais juste prendre le dossier "faso-dev" de mon repertoire "vendor (local)" et le mettre sur le serveur et faire un composer install ou update, cela va t-il passer??

**Comment by faso-dev** [comment]
Il vous déployer votre dossier vendor ou carrément faire un composer
install sur votre serveur

On Fri, Apr 29, 2022 at 09:27 AbdoulNhak ***@***.***> wrote:

> Pourtant en local tout marche très bien, vraiment je ne comprends rien,
> Pensez-vous que je devrais juste prendre le dossier "faso-dev" de mon
> repertoire "vendor (local)" et le mettre sur le serveur et faire un
> composer install ou update, cela va t-il passer??
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/faso-dev/orange-money-burkina-sdk/discussions/27#discussioncomment-2660259>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJTPVXRNLDMYBD5XLEA22XLVHOTOJANCNFSM5SGCY2QQ>
> .
> You are receiving this because you commented.Message ID:
> <faso-dev/orange-money-burkina-sdk/repo-discussions/27/comments/2660259@
> github.com>
>
-- 
Salutations cordiales,

*S.C jerôme ONADJA | Développeur web backend(PHP, JAVA, NODE JS)*

*Mob.* 51 00 65 57 - 56 02 90 91

*LinkedIn : https://www.linkedin.com/in/onass
<https://www.linkedin.com/in/onass> *
*Site web personnel | https://onass.herokuapp.com
<https://onass.herokuapp.com/>*

**Comment by AbdoulNhak** [comment]
Bonjour, voilà un extrait de code du controller :

try {
            $username= 'username';
            $password = 'password';
            $merchant_number ='76555585';
            $client_number = request('numero');
            $payment_amount = request('prix');
            $otp_code = request('otp');
            // $orangeApi = (new OrangeMoneyAPI("username", "password",
"merchant_number"))
            $orangeApi = (new OrangeMoneyAPI($username, $password,
$merchant_number))
                        ->withTransactionData(TransactionData::from(
$client_number, $payment_amount, $otp_code))
                        ->withCustomReference("123456778") //optionnal
                        ->useProdApi() // for production => useProdApi()
                        ->withoutSSLVerification() //if you have any
troubleshoot with ssl verifcation(not recommended)
            ;
            $response = (new PaymentSDK($orangeApi))->handlePayment();
